src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -249,15 +230,7 @@ func (r *StatefulSetResource) rollingUpdatePartition(
 func (r *StatefulSetResource) modifyPodImage(
 	stsSpec *corev1.PodSpec, newImage string,
 ) error {
-	if err := modifyContainerImage(stsSpec.InitContainers, configuratorContainerName, newImage); err != nil {
```

Comment:
We need to talk about the init container image and its versioning.

Reply:
I will restore the removed update process.

Reply:
Sorry this need to be removed. We can talk about it, but I would like to not stop the merging.

Reply:
I'm fine with temporarily removing this. Could you open an issue to ensure this comes back when we have Configurator image versioning? @RafalKorepta